### ISSUE TITLE
Is there a recommended validation library?

### ISSUE THREAD

**Issue by lcpriest** [martndemus/ember-form-for#94]
Ideally one that 'Just Works ™'.

**Comment by martndemus** [comment]
[ember-changeset](https://github.com/DockYard/ember-changeset) + [ember-changeset-validations](https://github.com/DockYard/ember-changeset-validations/) = It Just Works™

**Comment by lcpriest** [comment]
Would I just pass `(changeset user EmployeeValidations)` as the formObject and it works?

**Comment by martndemus** [comment]
If `EmployeeValidations` is a property on your component/controller, then yes!

**Comment by martndemus** [comment]
so:

```hbs
{{#form-for (changeset user EmployeeValidations) as |f|}}
   
{{/form-for}}
```

**Comment by lcpriest** [comment]
Thanks! I'll fiddle with it as I'm not seeing any errors pop up, but I imagine it's my own install as I've overridden the submit function

**Comment by lcpriest** [comment]
If a changeset has errors, `changeset.get('errors')` is an array - but it looks like the form-field tries to look them up as an object (with the attributeName as a key). re: https://github.com/martndemus/ember-form-for/blob/1524951c5972de4dcbc1129d85c26b45ca7178d1/addon/templates/components/form-field.hbs#L21

Am I missing something, or do I have to convert the array into an object for this to be read correctly?

**Comment by martndemus** [comment]
Hmm, you're right. Let me see what I can do to fix this!

**Comment by maprules1000** [comment]
👍  Seeing the same problem trying to use `ember-changeset` and `ember-changeset-validations`. 

There's actually 2 error properties on the changeset: `changeset.get('errors')` is an array of objects, but 'changeset.get('error')` is just an object with all validation errors in it.

**Comment by martndemus** [comment]
@maprules1000 & @lcpriest I published `2.0.0-alpha.7` that has added support for changing the property on which errors are lookup op on: https://github.com/martndemus/ember-form-for#ember-changeset
